TeleportEverything/DelayedAction.cs

```diff
@@ -0,0 +1,24 @@
+﻿using System;
+using System.Collections;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using UnityEngine;
+
+namespace TeleportEverything
+{
+    public class DelayedAction:MonoBehaviour
+    {
+        public void InvokeDelayed(System.Action aDelegate, float delay)
```

Comment:
I agree, coroutine is better here. Interestingly, the game uses a loop timer in this situation.

Reply:
btw I think we can test if the spawn time will be the same for both players.